pkg/operator/controllers/checker/checker_controller.go

```diff
@@ -59,7 +59,9 @@ func (r *CheckerController) Reconcile(request ctrl.Request) (ctrl.Result, error)
 		if thisErr != nil {
 			// do all checks even if there is an error
 			err = thisErr
```

Comment:
Is this necessary? Can't we use `err` instead of `thisErr`? The only real error-generating statement this this function is `c.Check(ctx)` so one variable should be enough

Reply:
good questions, why this is not enough?
```
for _, c := range r.checkers {
		err := c.Check(ctx)
		if err != nil {
			// do all checks even if there is an error
			if err != errRequeue {
				r.log.Errorf("checker %s failed with %v", c.Name(), err)
			}
		}
	}
```

Reply:
It's subtle.  Consider the following case: checker 0 returns an error, and checker 1 returns nil.  In that case by your code we would return foo, nil when we exit the function, but in fact we want to return foo, err.